libp2p/__init__.py

```diff
@@ -69,15 +95,22 @@ def initialize_default_swarm(
     :return: return a default swarm instance
     """
     # pylint: disable=too-many-arguments
+
+    if not id_opt:
+        id_opt = generate_id()
+
     if not swarm_opt:
         swarm_opt = initialize_default_swarm(
             id_opt=id_opt, transport_opt=transport_opt,
             muxer_opt=muxer_opt, sec_opt=sec_opt,
             peerstore_opt=peerstore_opt)
+    if not disc_opt:
+        disc_opt = initialize_default_kademlia(ksize=20, alpha=3,
+                                               id_opt=id_opt, storage=None)
 
     # TODO enable support for other host type
     # TODO routing unimplemented
-    host = BasicHost(swarm_opt)
+    host = RoutedHost(swarm_opt, disc_opt)
```

Comment:
I think we should still provide the option for only starting a BasicHost. Can we only return a BasicHost if no disc_opt is passed in and a RoutedHost only when a disc_opt is introduced?

Reply:
Fixed!